### ISSUE TITLE
Error messages (37)

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#345]
JSON:

`${param} : 2`

Result:

`DotDict({'param': 2})`

Expected: Error because of a parameter with name `"param"` does not exist.

Taken from

https://github.com/test-fullautomation/python-jsonpreprocessor/issues/260#issuecomment-2116979259

**Comment by HolQue** [comment]
The same with:

JSON:

```
"param"  : "value",
"params" : {"001" : {"002" : {
                              "param" : 1,
                              ${param} : 2
                             }
                    }
           }
```

Result:

```
{'param': 'value',
 'params': DotDict({'001': DotDict({'002': DotDict({'param': 2})})})}
```

`"param"` inside the dictionary has value 2. This is not expected because of a parameter with this name does not exist at this position. The scope is missing. Expected is the already known error message

`Error: 'A key with name '${param}' does not exist at this position. Use the '<name> : <value>' syntax to create a new key.'!`

This would be the correct way to overwrite an already existing parameter:

`${params.001.002.param} : 2`

Taken from

https://github.com/test-fullautomation/python-jsonpreprocessor/issues/260#issuecomment-2133073707

**Comment by HolQue** [comment]
Hi Son,

w.r.t. our discussion about https://github.com/test-fullautomation/python-jsonpreprocessor/issues/345#issue-2500927342

JSON:

`${param} : 2`

My proposal is to make the existence of at least one dot inside the expression a precondition for an implicit creation.

Outcome:

`${param} : 2 `is still an error (not existing parameter)

And this would be an implicit creation

`${par.am} : 2`

working well:

`DotDict({'par': DotDict({'am': 2})})`

**Comment by namsonx** [comment]
Hello Holger,

I implemented your proposal and pushed the change to the stabi branch.
Due to this change, there are some test cases of selftest in JPP package are failed:
```
 - JPP_0202-(PARAMETER_SUBSTITUTION)-[GOODCASE]
- JPP_0203-(PARAMETER_SUBSTITUTION)-[GOODCASE]
- JPP_0206-(PARAMETER_SUBSTITUTION)-[GOODCASE]
- JPP_0207-(PARAMETER_SUBSTITUTION)-[GOODCASE]
- JPP_1004-(IMPLICIT_CREATION)-[GOODCASE]
```
Could you please help me verify and update these test cases based on our new change?

Regarding the case in your second comment, it already implemented on the stabi branch.

Thank you,
Son

**Comment by HolQue** [comment]
OK, starting now with rework and maintenance of self test.

**Comment by HolQue** [comment]
Hi Son,

it's working fine now. And please rephrase the following error message:

`The parameter '${param}' at this position has to overwrite with the absolute path: ...`

Alternative:

`Missing scope for parameter '${param}'. To change the value of this parameter, an absolute path must be used: ...`

**Comment by namsonx** [comment]
Hello Holger,

Thank you for your suggestion! I updated and pushed to stabi branch.

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful. Issue can be closed.

**Comment by test-fullautomation** [comment]
merged for version 0.13.1
